lcls-twincat-common-components/Library/Devices/PPM/FB_PPM_PowerMeter.TcPOU

```diff
@@ -103,6 +172,26 @@ fbGetPMVoltage(
 fCalibBase := (fVoltage + fCalibRelOffset) * fCalibRelRatio;
 fCalibMJ := fCalibBase * fCalibMJRatio;
 
+// (Re)starting backround collection, will gracefully restart
+IF fBackgroundCollect = TRUE THEN
+    fBackgroundCollect := FALSE;
+    fBackgroundCollectionCycles := 10_000_000 * fBackgroundCollectionTime / _TaskInfo[GETCURTASKINDEXEX()].CycleTime;
```

Comment:
Any compiler warnings here for the implicit floating point -> int conversion? We can make it explicit to get rid of these via the `LREAL_TO_UDINT` function.